File: docs/guides/python/mcmSearchWithout.py
from algoliasearch.search.client import SearchClientSync


def _get_app_id_for(_user):
    # Implement your own logic here
    return ""


def _get_indexing_api_key_for(_user):
    # Implement your own logic here
    return ""


app_id = _get_app_id_for("user42")
api_key = _get_indexing_api_key_for("user42")

_client = SearchClientSync(app_id, api_key)

search_params = {
    "query": "<YOUR_SEARCH_QUERY>",
    "facetFilters": ["user:user42", "user:public"],
}

_client.search_single_index(
    index_name="<YOUR_INDEX_NAME>",
    search_params=search_params,
)

File: docs/guides/go/src/mcmSearchWithout.go
package main

import (
	"context"
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func mcmSearchWithout() {
	getAppIDFor := func(_ string) (string, error) {
		return "", nil // Implement your logic here
	}

	getIndexingAPIKeyFor := func(_ string) (string, error) {
		return "", nil // Implement your logic here
	}

	// Fetch from your own data storage and with your own code
	// the associated application ID and API key for this user
	appID, err := getAppIDFor("user42")
	if err != nil {
		fmt.Println(err)

		return
	}

	apiKey, err := getIndexingAPIKeyFor("user42")
	if err != nil {
		fmt.Println(err)

		return
	}

	client, err := search.NewClient(appID, apiKey)
	if err != nil {
		fmt.Println(err)

		return
	}

	searchParams := search.SearchParamsObjectAsSearchParams(
		search.NewSearchParamsObject().
			SetQuery("<YOUR_SEARCH_QUERY>").
			SetFacetFilters(
				search.ArrayOfFacetFiltersAsFacetFilters(
					[]search.FacetFilters{
						*search.StringAsFacetFilters("user:user42"),
						*search.StringAsFacetFilters("user:public"),
					},
				),
			),
	)

	_, err = client.SearchSingleIndex(context.Background(), "<YOUR_INDEX_NAME>", searchParams)
	if err != nil {
		panic(err)
	}
}
